algebra/tests/quat/various.c

```diff
@@ -932,3 +950,106 @@ TEST_GROUP_RUNNER(group_quat_quat2euler)
 	RUN_TEST_CASE(group_quat_quat2euler, quat_quat2euler_notUnitBiggerValues);
 	RUN_TEST_CASE(group_quat_quat2euler, quat_quat2euler_sourceRetain);
 }
+
+
+/* ##############################################################################
+ * -----------------------        quat_vecRot tests       -----------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_quat_vecRot);
+
+
+TEST_SETUP(group_quat_vecRot)
+{
+}
+
+
+TEST_TEAR_DOWN(group_quat_vecRot)
+{
+}
+
+
+TEST(group_quat_vecRot, quat_vecRot_baseQuaternions)
+{
+	quat_t qA, qI, qJ, qK, qZero = { 0 };
+	vec_t v1 = { .x = 1.0f, .y = 0.0f, .z = 0.0f };
+	vec_t v2 = { .x = -1.0f, .y = 0.0f, .z = 0.0f };
+	vec_t v, expected;
+
+	qA = qI = qJ = qK = qZero;
+	qA.a = qI.i = qJ.j = qK.k = 1.0f;
+
+	/* Rotation using qA quaternion. Nothing should change */
+	v = v1;
+	expected = v1;
+
+	quat_vecRot(&v, &qA);
+
+	TEST_ASSERT_EQUAL_VEC(expected, v);
+
+	/* Rotation using qI quaternion should be equal to rotation about 180 degrees along x-axis. Nothing should change */
+	v = v1;
+	expected = v1;
+
+	quat_vecRot(&v, &qI);
+
+	TEST_ASSERT_EQUAL_VEC(expected, v);
+
+	/* Rotation using qJ quaternion should be equal to rotation about 180 degrees along y-axis. */
+	v = v1;
+	expected = v2;
+
+	quat_vecRot(&v, &qJ);
+
+	TEST_ASSERT_EQUAL_VEC(expected, v);
+
+	/* Rotation using qK quaternion should be equal to rotation about 180 degrees along z-axis. */
+	v = v1;
+	expected = v2;
+
+	quat_vecRot(&v, &qK);
+
+	TEST_ASSERT_EQUAL_VEC(expected, v);
+}
+
+
+TEST(group_quat_vecRot, quat_vecRot_std)
+{
+	vec_t v = V1;
+
+	quat_vecRot(&v, &Q8);
+
+	TEST_ASSERT_EQUAL_VEC(V1rotQ8, v);
+}
+
+
+TEST(group_quat_vecRot, quat_vecRot_biggerValues)
+{
+	vec_t v = V2;
+
+	quat_vecRot(&v, &Q8);
+
+	TEST_ASSERT_EQUAL_VEC(V2rotQ8, v);
+}
+
+
+TEST(group_quat_vecRot, quat_vecRot_zeroVector)
+{
+	vec_t v1, v2 = { .x = 0.0f, .y = 0.0f, .z = 0.0f };
```

Comment:
Your idea for calling zero vectors `zeroVec` applies here as well:
```suggestion
	vec_t v1, zeroVec = { .x = 0.0f, .y = 0.0f, .z = 0.0f };
```